src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -196,10 +155,82 @@ public ApiClient load( Class<?> key ) throws Exception {
                 .stream()
                 .collect( ImmutableMap.toImmutableMap( PropertyType::getType, PropertyType::getId ) );
 
-        refreshStudyInformation();
+        appsDictionaryESID = entitySetsApi.getEntitySetId( USER_APPS_DICTIONARY );
+
         refreshUserAppsDictionary();
     }
 
+    private String getFirstValueOrNull( Map<FullQualifiedName, Set<Object>> entity, FullQualifiedName fqn ) {
+        if ( entity.getOrDefault( fqn, Set.of() ).isEmpty() ) {
+            return null;
+        }
+        return entity.get( fqn ).iterator().next().toString();
+    }
+
+    private UUID getFirstUUIDOrNull( Map<FullQualifiedName, Set<Object>> entity, FullQualifiedName fqn ) {
+        String firstValue = getFirstValueOrNull( entity, fqn );
+        if ( firstValue == null ) {
+            return null;
+        }
+
+        return UUID.fromString( firstValue );
+    }
+
+    private void initializeEntitySets( AppApi appApi, CollectionsApi collectionsApi ) {
+
+        // get mapping appName -> appId
+        Map<String, UUID> appNameIdMap = APP_NAMES.stream()
+                .map( appApi::getAppByName )
+                .collect( Collectors.toMap( App::getName, AbstractSecurableObject::getId ) );
+
+        Map<String, Map<UUID, Map<String, UUID>>> entitySets = new HashMap<>();
+
+        // get configs for each app
+        appNameIdMap.forEach( ( appName, appId ) -> {
+            List<UserAppConfig> configs = appApi.getAvailableAppConfigs( appId );
+
+            Map<UUID, Map<String, UUID>> orgEntitySetMap = new HashMap<>();
+
+            configs.forEach( userAppConfig -> {
+                EntitySetCollection entitySetCollection = collectionsApi
+                        .getEntitySetCollection( userAppConfig.getEntitySetCollectionId() );
+                Map<UUID, UUID> template = entitySetCollection.getTemplate();
+
+                Map<String, UUID> templateNameEntitySetId = collectionsApi
+                        .getEntityTypeCollection( entitySetCollection.getEntityTypeCollectionId() )
+                        .getTemplate()
+                        .stream()
+                        .collect( Collectors.toMap( CollectionTemplateType::getName,
+                                templateType -> template.get( templateType.getId() ) ) );
+
+                orgEntitySetMap.put( userAppConfig.getOrganizationId(), templateNameEntitySetId );
+            } );
+
+            entitySets.put( appName, orgEntitySetMap );
+        } );
+
+        entitySetIdsByOrgId = ImmutableMap.copyOf( entitySets );
+    }
+
+    private UUID getEntitySetId( UUID organizationId, String appName, String templateName ) {
+        Map<String, UUID> templateEntitySetIdMap = entitySetIdsByOrgId
+                .getOrDefault( appName, ImmutableMap.of() )
+                .getOrDefault( organizationId, ImmutableMap.of() );
+
+        if ( templateEntitySetIdMap.isEmpty() ) {
```

Comment:
This seems like we should raise an exception and fail.

Reply:
Actually we shouldn't raise an exception. There are use cases where we should return a null if the orgId -> template -> entitySetId mapping is not found). For example in the `deleteStudyData` function https://github.com/openlattice/chronicle-server/blob/2b3a2f8740617d61c39446a257a6a5427ea26d96/src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java#L1150 there are entity set ids which will be null if the corresponding app module is not installed for a particular organization. Since this is expected we shouldn't raise an exception. However, for most use cases, there is a `Preconditions.checkNotNull` wrapper (https://github.com/openlattice/chronicle-server/blob/2b3a2f8740617d61c39446a257a6a5427ea26d96/src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java#L688).